src/config.cpp

```diff
@@ -248,8 +244,8 @@ struct VisionOutputs_Element : JSON::Element {
   explicit VisionOutputs_Element(Config::Model::Vision::Outputs& v) : v_{v} {}
 
   void OnString(std::string_view name, std::string_view value) override {
-    if (name == "visual_features") {
-      v_.visual_features = value;
+    if (name == "image_features") {
```

Comment:
Will this change break the old models?

Reply:
It will and it is intentional. The old design is quite restrictive in terms of both obtaining and running the vision models. The new design makes it easier to obtain and run the vision models. It also allows users to run Phi-3 vision and Phi-3.5 vision with multiple images, which was not previously supported in the uploaded ONNX models as well as ONNX Runtime GenAI. These changes also make the multi-modal pipeline more generic to onboard new vision models.

Since backwards compatibility is not fully guaranteed, it should be fine to make this change. We will publish updated ONNX models for Phi-3 vision along with the Phi-3.5 vision ONNX models.